dbloader/riak_loader.py

```diff
@@ -95,31 +99,66 @@ def update(self, bucket, table=None, custom=None):
             return False
         return time.time() - start_time
 
-    def delete(self, bucket, table, custom=None):
+    def delete(self, bucket_name, table, custom=None):
         '''
         Delete a single object
         '''
+
         start_time = time.time()
         if not self.conn:
             self.get_connection()
         try:
-            b = self.conn.bucket(bucket)
-            result = b.delete(str(custom))
+            b = self.conn.bucket(bucket_name)
+            obj = b.get(str(custom))
+            r = obj.delete()
 
-        except Exception:
-            logger.exception('Unable to delete key')
+            logger.debug('Deleted %s from %s', custom, bucket_name)
+
+        except Exception as e:
+            logger.exception('Unable to delete key %s: %s', custom, e)
             raise Exception
         return time.time() - start_time
 
-    def select(self, bucket, table=None, custom=None):
+    def truncate(self, bucket_name, table=None, custom=None):
+        '''
+        Truncate a table/collection/bucket
+        '''
+        logger.debug('Truncating %s with %s', bucket_name, custom)
+        tbucket = self.conn.bucket(bucket_name)
+        keycount = 0
+        get_one = 0
+        kfile = self.keyfile + ".{0}".format(bucket_name)
+        with open(kfile, "w") as myFile:
+            try:
+                with closing(self.conn.stream_keys(tbucket, self.timeout)) as keys:
+                    for key_list in keys:
+                        for key in key_list:
+                            keycount = keycount + 1
+                            myFile.write("%s\n" % key)
```

Comment:
Are we planning on using/keeping this key file as a record of what keys we deleted? Will it make sense to combine the two process: get a key and delete a key in single for like this:
``` 
with closing(self.conn.stream_keys(tbucket, self.timeout)) as keys:
    for key_list in keys:
         for key in key_list:
             keycount = keycount + 1
             logger.debug("deleting key: %s" % key)
             deletes.append(tpool.spawn(self.delete, bucket_name, None, key.strip()))
tpool.join()
```

This will save us from not enough space error in case of huge buckets and low temp space while writing and reading from file.

Reply:
The initial plan was to support an option to dump results into s3 as a backup, but figured that could be in a separate PR.  That said, this way seems good until that time, especially since it would still be optional.
Does that sound fine?  ( Loader.delete() having a option to save to s3

Reply:
sounds good 👍